test/test_prototype_features.py

```diff
@@ -0,0 +1,170 @@
+        elif format == features.BoundingBoxFormat.XYWH:
+            x = torch.randint(0, width // 2, ())
+            y = torch.randint(0, height // 2, ())
+            w = torch.randint(1, width - int(x), ())
+            h = torch.randint(1, height - int(y), ())
+            parts = (x, y, w, h)
+        elif format == features.BoundingBoxFormat.CXCYWH:
+            cx = torch.randint(1, width - 1, ())
+            cy = torch.randint(1, height - 1, ())
+            w = torch.randint(1, min(int(cx), width - int(cx)), ())
+            h = torch.randint(1, min(int(cy), height - int(cy)), ())
+            parts = (cx, cy, w, h)
+
+        return features.BoundingBox.from_parts(*parts, format=format, image_size=image_size)
+
+    @pytest.mark.parametrize(("format", "intermediate_format"), itertools.permutations(("xyxy", "xywh"), 2))
+    def test_cycle_consistency(self, format, intermediate_format):
+        input = self.make_bounding_box(format=format)
+        a = input.convert(intermediate_format)
+        output = a.convert(format)
+        assert_close(input, output)
+
+
+class TestJit:
+    def test_bounding_box(self):
+        def resize(input: features.BoundingBox, size: torch.Tensor) -> features.BoundingBox:
+            old_height, old_width = input.image_size
+            new_height, new_width = size
+
+            height_scale = new_height / old_height
+            width_scale = new_width / old_width
+
+            old_x1, old_y1, old_x2, old_y2 = input.convert("xyxy").to_parts()
+
+            new_x1 = old_x1 * width_scale
+            new_y1 = old_y1 * height_scale
+
+            new_x2 = old_x2 * width_scale
+            new_y2 = old_y2 * height_scale
+
+            return features.BoundingBox.from_parts(
+                new_x1, new_y1, new_x2, new_y2, like=input, format="xyxy", image_size=tuple(size.tolist())
+            )
+
+        def horizontal_flip(input: features.BoundingBox) -> features.BoundingBox:
+            x, y, w, h = input.convert("xywh").to_parts()
+            x = input.image_size[1] - (x + w)
+            return features.BoundingBox.from_parts(x, y, w, h, like=input, format="xywh")
+
+        def compose(input: features.BoundingBox, size: torch.Tensor) -> features.BoundingBox:
+            return horizontal_flip(resize(input, size)).convert("xyxy")
+
+        image_size = (8, 6)
+        input = features.BoundingBox([2, 4, 2, 4], format="cxcywh", image_size=image_size)
+        size = torch.tensor((4, 12))
+        expected = features.BoundingBox([6, 1, 10, 3], format="xyxy", image_size=image_size)
+
+        actual_eager = compose(input, size)
+        assert_close(actual_eager, expected)
+
+        sample_inputs = (features.BoundingBox(torch.zeros((4,)), image_size=(10, 10)), torch.tensor((20, 5)))
+        actual_jit = torch.jit.trace(compose, sample_inputs, check_trace=False)(input, size)
+        print(actual_jit)
+        print(expected)
+        assert_close(actual_jit, expected)
```

Comment:
This test fails, but I was unable to get it to pass. The eager variant works as expected, but the jitted one gives incorrect output. I'm not sure if the test setup is wrong somehow or of JIT simply doesn't work with tensor subclasses that carry additional meta data. This needs to be reviewed by someone with more JIT experience than I have. cc @fmassa @datumbox

Reply:
Pinging @Chillee here, but my gut feeling is that attributes of the Tensor subclass might be embedded in the graph as constants.

Reply:
You can check the computation graph of the traced model (via `print(traced_function.code)`) to see if the `image_size` was embedded as a constant or not.

Another thing would be to make the `image_size` for `sample_inputs` to be the same of for the `input`, which should make tests pass (but it is still a bug)

Reply:
You are probably right:

```python
def compose(input: Tensor,
    size: Tensor) -> Tensor:
  new_height, new_width, = torch.unbind(size)
  height_scale = torch.div(new_height, CONSTANTS.c0)
  width_scale = torch.div(new_width, CONSTANTS.c0)
  data = torch.clone(input)
  data0 = torch.to(data, torch.device("cpu"), 6)
  old_x1, old_y1, old_x2, old_y2, = torch.unbind(torch.detach(data0), -1)
  a = torch.mul(old_x1, width_scale)
  b = torch.mul(old_y1, height_scale)
  c = torch.mul(old_x2, width_scale)
  d = torch.mul(old_y2, height_scale)
  data1 = torch.stack([a, b, c, d], -1)
  data2 = torch.to(data1, torch.device("cpu"), 6)
  x, y, x2, y2, = torch.unbind(torch.detach(data2), -1)
  c0 = torch.sub(x2, x)
  d0 = torch.sub(y2, y)
  data3 = torch.stack([x, y, c0, d0], -1)
  data4 = torch.to(data3, torch.device("cpu"), 6)
  x0, b0, w, d1, = torch.unbind(torch.detach(data4), -1)
  a0 = torch.rsub(torch.add(x0, w), 5)
  data5 = torch.stack([a0, b0, w, d1], -1)
  data6 = torch.to(data5, torch.device("cpu"), 6)
  x1, y1, w0, h, = torch.unbind(torch.detach(data6), -1)
  c1 = torch.add(x1, w0)
  d2 = torch.add(y1, h)
  data7 = torch.stack([x1, y1, c1, d2], -1)
  data8 = torch.to(data7, torch.device("cpu"), 6)
  return torch.detach(data8)
```

The important lines are

```python
new_height, new_width, = torch.unbind(size)
height_scale = torch.div(new_height, CONSTANTS.c0)
width_scale = torch.div(new_width, CONSTANTS.c0)
```

which come from 

```python
old_height, old_width = input.image_size
new_height, new_width = size

height_scale = new_height / old_height
width_scale = new_width / old_width
```

So `old_height` and `old_width`, which come directly from the `BoundingBox.image_size` meta data, are stored as constants.

Reply:
Here is what I propose: let's forget torchscript for now and continue with the prototype